tests/ClientTest.php

```diff
@@ -197,6 +197,21 @@ public function testAgreement() {
         self::assertEquals($httpClient, $client->getHttpClient());
     }
 
+    /**
+     * Test that Client automatically generates the correct HttpClient if one
+     * is not provided.
+     */
+    public function testProvisionHttpClient() {
+        $expectedHttpClient = new HttpClient([
+            'base_uri' => 'https://api.smarteru.com/apiv2/'
+        ]);
+
+        $client = new Client();
+        $httpClient = $client->getHttpClient();
+        self::assertInstanceOf(HttpClient::class, $httpClient);
+        self::assertEquals($httpClient, $expectedHttpClient);
```

Comment:
Thanks for providing a test, good call.

Food for thought... What's importance is that the return value is an `HttpClient` and that the base URI is set correctly not that it equals the test instance. You could more concisely test this as:

```suggestion
        $client = new Client();
        $httpClient = $client->getHttpClient();
        self::assertInstanceOf(HttpClient::class, $httpClient);
        self::assertEquals(Client::POST_URL, $httpClient->config['base_uri']);
```